src/services/catalog.service.ts

```diff
@@ -35,8 +38,46 @@ export class CatalogService {
         }
       });  // .ea category
     });  // .ea tag
+    if (_.isEmpty(catsBySubCats)) {
+      catsBySubCats[otherId] = otherId;
+    }
     return catsBySubCats;
   }
+
+  public hasCategory(item: any, category: string) {
+    return _.includes(item.catsBySubCats, category);
+  }
+
+  public hasSubCategory(item: any, subCategory: string) {
+    return _.includes(Object.keys(item.catsBySubCats), subCategory);
+  }
+
+  /**
+   * Return a new Array of only those Categories and SubCategories which
+   * exist in the passed in items.
+   * @param items
+   * @returns {Array}
+   */
+  public removeEmptyCategories(items: IServiceItem) {
```

Comment:
Sounds good. Doesn't have to be this PR.

@cshinn just a heads up that the markup and CSS is going to be tricky for the layout we want. @rhamilto and I looked at this and couldn't find a clean solution in using just CSS and HTML. We had something basic working, though, if we made the expanded section a fixed height.